docs/rules/no-module-extensions.md

```diff
@@ -0,0 +1,29 @@
+# suitescript/no-module-extensions
+
+Enforces no filename extensions on module dependencies.
+
+There appears to be a NetSuite bug that intermittently causes an error (`SuiteScript 2.1 entry point scripts must implement one script type function..`) when attempting to upload script files that include dependencies with file extensions.
```

Comment:
This is great info to include. I think since it's a temporary bug that will be fixed in the future (hopefully), we should add a date to mark when this was written.

```suggestion
As of January 2021, there appears to be a NetSuite bug that intermittently causes an error (`SuiteScript 2.1 entry point scripts must implement one script type function..`) when attempting to upload script files that include dependencies with file extensions.
```